### ISSUE TITLE
WARN Duplicate AssetLoader preregistered for Asset type `bevy_image::image::Image` with extensions `["hdr"]`. when run breakout example

### ISSUE THREAD

**Issue by C-Entropy** [bevyengine/bevy#15730]
## Bevy version

The release number or commit hash of the version you're using.

```
git log -1
commit 4bf647ff3b0ca7c8ca47496db9cfe03702328473 (HEAD -> main, origin/main, origin/HEAD)
```

## \[Optional\] Relevant system information

If you cannot get Bevy to build or run on your machine, please include:

- the Rust version you're using (you can get this by running `cargo --version`)
```
cargo --version
cargo 1.81.0 (2dbb1af80 2024-08-20)
```
  - Bevy relies on the "latest stable release" of Rust
  - nightly should generally work, but there are sometimes regressions: please let us know!
- the operating system or browser used, including its version
  - Windows 11 23H2

If your bug is rendering-related, copy the adapter info that appears when you run Bevy.

```ignore
`INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce RTX 4050 Laptop GPU", vendor: 4318, device: 10465, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "560.94", backend: Vulkan }`
```

You should also consider testing the examples of our upstream dependencies to help isolate any setup-specific issue:

- [`wgpu`](https://github.com/gfx-rs/wgpu) for rendering problems
- [`winit`](https://github.com/rust-windowing/winit) for input and window management
- [`gilrs`](https://docs.rs/gilrs/latest/gilrs/) for gamepad inputs

## What you did
 ` cargo run --example breakout`

## What went wrong

If it's not clear, break this out into:

- what were you expecting?
- Run successfully
- what actually happened?

```
cargo run --example breakout
...
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 51.48s
     Running `target\debug\examples\breakout.exe`
2024-10-08T07:58:04.636297Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows 11 Home China", kernel: "22631", cpu: "Intel(R) Core(TM) i7-14650HX", core_count: "16", memory: "15.7 GiB" }
2024-10-08T07:58:05.580350Z  INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce RTX 4050 Laptop GPU", vendor: 4318, device: 10465, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "560.94", backend: Vulkan }
2024-10-08T07:58:05.701200Z  WARN bevy_asset::server::loaders: Duplicate AssetLoader preregistered for Asset type `bevy_image::image::Image` with extensions `["hdr"]`. Loader must be specified in a .meta file in order to load assets of this type with these extensions.
2024-10-08T07:58:06.496605Z  INFO bevy_winit::system: Creating new window "App" (0v1#4294967296)
2024-10-08T07:58:06.585224Z  INFO breakout::stepping: Bevy was compiled without stepping support. Run with `--features=bevy_debug_stepping` to enable stepping.
2024-10-08T07:58:07.158131Z  WARN bevy_ecs::system::function_system: bevy_pbr::render::mesh_view_bindings::prepare_mesh_view_bind_groups did not run because it requested inaccessible system parameter Res<OitBuffers>
2024-10-08T07:58:11.091470Z  INFO bevy_window::system: No windows are open, exiting
2024-10-08T07:58:11.094472Z  INFO bevy_winit::system: Closing window 0v1#4294967296
2024-10-08T07:58:11.124012Z  INFO bevy_window::system: No windows are open, exiting
```

## Additional information

Other information that can be used to further reproduce or isolate the problem.
This commonly includes:

- screenshots
- logs
- theories about what might be going wrong
- workarounds that you used
- links to related bugs, PRs or discussions

**Comment by tychedelia** [comment]
Related to #15650?

**Comment by rparrett** [comment]
bisected to #15586

**Comment by andriyDev** [comment]
I have a fix for this, but it's blocked on #15812.

For the context: In #15586, we added feature gates for all the image formats. We also moved all the supported extension stuff to `bevy_image`. This included making a constant array of all the supported formats. Sadly, this list of supported formats is on the `ImageFormat` type, not the `ImageLoader` type. This makes it awkward since we can't remove exr and hdr formats from this list (because then the array doesn't have all the formats like it suggests). We also can't move this list of supported formats to `bevy_render`, since it doesn't have all the image format feature flags. So #15812 moves the `ImageLoader` to `bevy_image`, and my next PR moves the supported format stuff to `ImageLoader` and removes exr and hdr from that list.
